src/socket/socket.cr

```diff
@@ -33,16 +84,61 @@ lib C
   end
 
   fun htons(n : Int32) : Int16
-  fun bind(fd : Int32, addr : SockAddrIn*, addr_len : Int32) : Int32
+  fun bind(fd : Int32, addr : SockAddr*, addr_len : Int32) : Int32
   fun listen(fd : Int32, backlog : Int32) : Int32
-  fun accept(fd : Int32, addr : SockAddrIn*, addr_len : Int32*) : Int32
-  fun connect(fd : Int32, addr : SockAddrIn*, addr_len : Int32) : Int32
+  fun accept(fd : Int32, addr : SockAddr*, addr_len : Int32*) : Int32
+  fun connect(fd : Int32, addr : SockAddr*, addr_len : Int32) : Int32
   fun gethostbyname(name : UInt8*) : HostEnt*
+  fun getsockname(fd : Int32, addr : SockAddr*, addr_len : Int32*) : Int32
+  fun getpeername(fd : Int32, addr : SockAddr*, addr_len : Int32*) : Int32
   fun setsockopt(sock : Int32, level : Int32, opt : Int32, optval : Void*, optlen : Int32) : Int32
 
   SOCK_STREAM = 1
+  SOCK_DGRAM = 2
+  SOCK_RAW = 3
+
+  IPPROTO_IP = 0
+  IPPROTO_TCP = 6
+  IPPROTO_UDP = 17
+  IPPROTO_RAW = 255
+
   SOL_SOCKET = 0xffff
   SO_REUSEADDR = 0x0004
+
+  INET_ADDRSTRLEN = 16
+  INET6_ADDRSTRLEN = 46
+end
+
+class SocketError < Exception
+end
+
+class Socket < FileDescriptorIO
+  def afamily(family)
```

Comment:
this method is required because the C addrinfo struct is an Int32 whereas the constants and socket methods take a UInt16 or UInt8 (depending on platform).